src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const currentArrayLength = this.length;
+    const joinSeparator = String(separator);
+    let resultingString = '';
+
+    if (!currentArrayLength) {
+      return resultingString;
+    }
+
+    this.forEach((item, index) => {
+      let stringCharacter = item;
+
+      if (item === null || item === undefined) {
+        stringCharacter = '';
+      }
+
+      if (currentArrayLength - 1 === index) {
+        resultingString += stringCharacter;
+      } else {
+        resultingString += stringCharacter + joinSeparator;
+      }
```

Comment:
Do we really need `else` here? Maybe we can simplify this and use only `if`?
P.S. We do the same part of logic in `if` and in `else` block